### ISSUE TITLE
Semantic hightlighting not available for Python in Jupyter notebook.

### ISSUE THREAD

**Issue by BrtXa** [microsoft/vscode-jupyter#15306]
Open a jupyter notebook (`.ipynb`) and type anything. The variables will be in white instead of being highlighted.

Note that semantic highlighting is still working normally in Python scripts (`.py`) files. 

## Environment data

-   VS Code version: 1.87
-   Jupyter Extension version (available under the Extensions sidebar): 2024.2.0
-   Python Extension version (available under the Extensions sidebar): 2024.2.1
-   OS (Windows | Mac | Linux distro) and version: Debian 12 and Ubuntu 22.04 both faced the same issue. 
-   Python and/or Anaconda version: 3.9.18 (but the same issue happens with Python >=3.9).
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): venv, tested with conda, same issue.
-   Jupyter server running: Local | Remote | Local

## Expected behaviour

Python code variables are correctly highlighted.

## Actual behaviour

Python code variables are not highlighted at all.

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. XXX

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by DonJayamanne** [comment]
> VS Code version: XXX
Jupyter Extension version (available under the Extensions sidebar): XXX
Python Extension version (available under the Extensions sidebar): XXX
OS (Windows | Mac | Linux distro) and version: XXX
Python and/or Anaconda version: XXX

In order tophelp you, we request you provide the information requested in three issues templates 

Also do you have the pylance extension installed, if you do then what version

**Comment by jnooree** [comment]
Same issue here.

VS Code version: XXX
Jupyter Extension version (available under the Extensions sidebar): XXX
Python Extension version (available under the Extensions sidebar): XXX
OS (Windows | Mac | Linux distro) and version: XXX
Python and/or Anaconda version: XXX

**Comment by jnooree** [comment]
Maybe related to microsoft/vscode#206487? I had a similar issue and it is now resolved after updating to v1.87.1.

**Comment by BrtXa** [comment]
> > VS Code version: XXX
> > Jupyter Extension version (available under the Extensions sidebar): XXX
> > Python Extension version (available under the Extensions sidebar): XXX
> > OS (Windows | Mac | Linux distro) and version: XXX
> > Python and/or Anaconda version: XXX
> 
> In order tophelp you, we request you provide the information requested in three issues templates
> 
> Also do you have the pylance extension installed, if you do then what version


Hi, I have updated my issue with missing information, I do have Pylance 2024.2.3 installed on my system.

Here are some screenshots to illustrate my issue better. 

On notebook:
![Screenshot from 2024-03-08 12-16-50](https://github.com/microsoft/vscode-jupyter/assets/72955139/32bd308e-b245-4a58-bd9e-79ff195ba585)

On a normal Python script:
![Screenshot from 2024-03-08 12-18-32](https://github.com/microsoft/vscode-jupyter/assets/72955139/45ca6273-c3a2-4e1e-967c-f0d75048d641)

I also suspect there might be some crashes between extensions. But note that this only happens on Jupyter Notebook. I faced this issue on 2 computers, One runs Linux locally and the other is Windows with WSL installed.
